### PR TITLE
Add API to support credential update

### PR DESCRIPTION
**Pull request by talakad**
Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/server/sdk/credentials.go [hunk 1]
```diff
@@ -589,3 +589,194 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
...
+	if len(req.GetBucket()) != 0 {
+		params[api.OptCredBucket] = req.GetBucket()
+	}
+	if len(aws.GetRegion()) != 0 {
+		params[api.OptCredRegion] = aws.GetRegion()
+	}
+	if len(aws.GetEndpoint()) != 0 {
+		params[api.OptCredEndpoint] = aws.GetEndpoint()
+	}
+	if len(aws.GetAccessKey()) != 0 {
```
Comment (ram-infrac) [inline api/server/sdk/credentials.go:638]: should we put additional checks before calling driver update for missing access/secrets ?

**Review by ram-infrac — APPROVED** [review]
lgtm

File: api/server/sdk/credentials.go (hunk 1 above)
Comment (talakad) [reply api/server/sdk/credentials.go:638]: It would be good for server to handle those. Hence not included here.

Comment (adityadani) [reply api/server/sdk/credentials.go:638]: Isn't this server itself ? I see the checks on google but not for aws and azure

File: api/server/sdk/credentials.go [hunk 2]
```diff
@@ -589,3 +589,194 @@ func validateAndDeleteIfInvalid(ctx context.Context, s *CredentialServer, uuid s
...
+
+func (s *CredentialServer) googleUpdate(
+	ctx context.Context,
+	req *api.SdkCredentialCreateRequest,
+	google *api.SdkGoogleCredentialRequest,
+	credId string,
+) (*api.SdkCredentialUpdateResponse, error) {
+
+	if len(google.GetJsonKey()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Must supply JSON Key")
```
Comment (adityadani) [inline api/server/sdk/credentials.go:710]: Why are we not doing the same checks for azure and aws ?

**Review by adityadani — APPROVED** [review]
looks good

File: api/server/sdk/credentials.go (hunk 2 above)
Comment (talakad) [reply api/server/sdk/credentials.go:710]: Thanks!, actually these checks are not needed. Users need not provide these, if they dont want to modify them. I should take these checks out.

File: api/server/sdk/credentials.go (hunk 1 above)
Comment (talakad) [reply api/server/sdk/credentials.go:638]: What I meant is combination of these fields for an update the implementation can decide to check on in whatever way they want. Here we dont need to check as not all of these fields are mandatory for update.

**Pull request merged**